### DISCUSSION TITLE
电脑开机后第一次延迟测速（或手动 TCPing）会出现延迟明显偏高，但后续则恢复正常的情况，大家有没有遇到过？

### DISCUSSION THREAD

**Discussion by XIU2** [XIU2/CloudflareSpeedTest#306] [讨论]
说一下我一直以来都发现的一个现象。
每次电脑开机后，无论是 CloudflareST 的延迟测速，还是手动 TCPing 都会发现，**第一次延迟明显偏高**，但从**第二个开始就基本恢复正常**了（手动 TCPing 时可以看到第二个明显延迟降低了一截，而 CloudflareST 因为是短时间内**多线程高并发**测速，所以平均延迟也会明显偏高），我也搞不懂是什么原因。

> 另外，我发现把 `tcping.exe` 复制到其他文件夹后（或重命名），首次运行也会出现第一次延迟明显偏高的现象，也就是说只要文件路径（包括文件名）改变了，那么首次 TCPing 链接都会延迟明显偏高。

如果只是 CloudflareST 出现这种情况，可能是代码问题，但手动 TCPing 也这样，而且文件位置/文件夹变了也会再次出现，那多半是**系统底层机制之类**的问题，这就触及到我的知识盲区了，有没有懂的？

另外，也不知道其他系统是否存在该问题？

搞得我每次都要先**随便延迟测速几个 IP**，后续再测速就没问题了（直到关机前都不会再出现了）。

****

### \# 临时解决方法

电脑开机后，第一次正式测速前，**先随便测几个 IP**（无需等待延迟测速完成，只要进度条动了就可以直接关了，比如你可以双击 `CloudflareST.exe` 等到软件显示出来进度条后就可以关了）。

**Comment by charSLee013** [comment]
斗胆做个假设，就当作抛砖引玉了

----

假设前提是 *每次电脑开机后* 是指 **网络活动静止一段时间后** 
第一次连接 *Cloudflare* 网段速度较慢

有可能是路由表被刷新了
比如在**第一次测速**的时候，IP 包在路由之间交换的时候，是需要更新路由表
总的时间就变成了：**IP包路由时间**+ **路由表查询时间** 

所以会看到第一次连接的时间是显著高于平均值的
而后因为网络活动活跃了，路由表上的Cloudflare IP段的Live Time 能保持较高的优先级而不会被路由刷掉了

**Comment by XIU2** [comment]
@charSLee013 不太懂这方面，不过该问题并不仅限于 Cloudflare CDN 的 IP，因为我平时还会去测速 / 手动 TCPing 其他的 IP。
